### DISCUSSION TITLE
3D Reconstruction of Isometric Sprites

### DISCUSSION THREAD

**Discussion by Praytic** [Praytic/youtd2#105] [Ideas]
The objective is to expedite the transition of future versions of You TD to a 3D environment using generative AI models. At present, all tower and item sprites are created with prompt2image tools. However, these tools are unable to render sprites with different perspectives, and they certainly can't generate 3D models. This discussion is aimed at exploring tools that can facilitate the automatic conversion of isometric sprites into static 3D models.

### Generated 3D models

https://www.kaedim3d.com/ - image2model offering
https://www.hypothetic.art/ 
https://www.mirageml.com/ - build a scene out of 3D assets

### Structure from motion

https://em3dscanningapp.com/ - free ios app
https://alicevision.org/#meshroom https://github.com/alicevision/Meshroom - open source software. Can create a cloud point using a set of photos (or frames from video), no additional software required, but it takes hours to process. Some parameters exist to speed up the process
https://www.capturingreality.com/ - free software, but you cannot export anything. The same as the previous one
https://github.com/OpenDroneMap/ODM - open source software to build clouds based on drone photos. Probably doesn’t work well with small objects
https://github.com/cvg/Hierarchical-Localization - open source, has some sfm pipeline.

### Links:
https://towardsdatascience.com/the-ultimate-guide-to-3d-reconstruction-with-photogrammetry-56155516ddc4

**Comment by wangluodaxia** [comment]
When will the 3D version be make?

**Reply by Kvel2D** [reply]
Hello, there are no plans for 3D graphics currently.
